### PR TITLE
Docs: Instructions on how to write E2E tests for app plugins

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
This PR provides some instructions on how to structure your E2E tests for app plugins. Hopefully it show case two different patterns for improving readability and to make it easier to share logic between tests. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This is a rough draft so feel free to provide bigger text changes that would make it easier to understand the documentation.

### CODE REVIEW

**Review by josmperez — COMMENTED** [review]
Hi @mckn - Please take a look at my suggestions in my recent commits related to doc style, clarity, and so on. Many of these are related to doc style preference for simple indicative tense and avoidance of gerunds. Let me know if you have any questions.

**Comment by mckn** [comment]
> Hi @mckn - Please take a look at my suggestions in my recent commits related to doc style, clarity, and so on. Many of these are related to doc style preference for simple indicative tense and avoidance of gerunds. Let me know if you have any questions.

Love it! There is only one thing that I would like to change. I'm adding a comment with your name to it so we can discuss it in this PR. Hope that is ok 🙏🏻

File: docusaurus/docs/e2e-test-a-plugin/test-an-app-plugin/app-pages.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  await gotoPage('/start');
+  await expect(page.getByRole('heading', { name: 'Welcome to my app' })).toBeVisible();
+});
+```
+
+## Page with a complex UI
+
+If, on the other hand, you have a page with a complex UI that might not be easy to interact with via the standard Playwright APIs, then you need a different approach. As a best practice, create a page object that can encapsulate the page's logic in functions.
+
+This is especially helpful if you want to reuse that selector logic in multiple tests. The example below showcases how to extend the standard fixture with a page object. Simplify the `getWelcomeText` logic because you want to highlight the page object pattern rather than write a complex selector logic.
```
Comment (mckn) [inline docusaurus/docs/e2e-test-a-plugin/test-an-app-plugin/app-pages.md:66]:
```suggestion
This is especially helpful if you want to reuse that selector logic in multiple tests. The example below showcases how to extend the standard fixture with a page object. We have simplified the `getWelcomeText` logic to highlight the page object pattern without introducing unnecessary complexity to this example.
```

Comment (mckn) [reply docusaurus/docs/e2e-test-a-plugin/test-an-app-plugin/app-pages.md:66]: What do you think about something like this @josmperez

**Review by josmperez — APPROVED** [review]
LGTM.

**Review by sunker — APPROVED** [review]
Awesome work on this @mckn!

**Pull request merged**